nixos/infrastructure/flyingcircus-virtual.nix

```diff
@@ -83,12 +83,24 @@ mkIf (cfg.infrastructureModule == "flyingcircus") {
       fsType = "xfs";
     };
     "/tmp" = {
-      device = "/dev/disk/by-label/tmp";
+      device = "/dev/disk/by-partlabel/tmp";
       fsType = "xfs";
       noCheck = true;
     };
   };
 
+  flyingcircus.initrd = {
+    upgradeXFS = {
+      "/dev/disk/by-label/root" = [ "bigtime=1" ];
+    };
+    # XXX: remove nrext=0 once all VMs are on kernel 6.6+
+    formatXFS = {
+      "/dev/disk/by-partlabel/tmp" =
+        "-L tmp -q -K -m crc=1,finobt=1,bigtime=1 -i nrext64=0 -d su=4m,sw=1";
+    };
```

Comment:
Is there a reason to do this here in fc-nixos instead of in fc.qemu? As far as I understand, the tmp partition is reformatted on every (cold) boot by fc.qemu, so doing it again here in the initrd as well would be redundant.

Additionally, `bigtime=1` has been set by default in mkfs.xfs for a few years now (compare the settings files for kernels 5.10 and 5.15 in xfsprogs), so this might have fixed itself already without us needing to do anything. I'm fairly sure this bit us once already with an old VM after upgrading the hardware from 21.05 to 24.11.

(Speaking of hardware, we might want to set `flyingcircus.initrd.upgradeXFS` on physical machines so that they get the filesystem upgrade as well.)

Reply:
The idea is that `/tmp` isn't reformatted on every boot cycle in fc.qemu. We don't know the kernel version inside a VM in fc.qemu, so this is quite error-prone. Here we know the kernel version of the VM, so this allows us to use new XFS features more agile without waiting for the full fleet to be on a newer kernel.

Reply:
So also this will be the primary place XFS flags are set